SwiftCompilerSources/Sources/Basic/SourceLoc.swift

```diff
@@ -0,0 +1,73 @@
+//===--- SourceLoc.swift - SourceLoc bridiging utilities ------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public struct SourceLoc {
+  public var pointer: UnsafePointer<UInt8>
+
+  public init?(pointer: UnsafePointer<UInt8>?) {
+    guard let pointer = pointer else {
+      return nil
+    }
+    self.pointer = pointer
+  }
+
+  public init?(pointer: UnsafePointer<CChar>?) {
+    guard let pointer = pointer else {
+      return nil
+    }
+    self.pointer = UnsafeRawPointer(pointer).assumingMemoryBound(to: UInt8.self)
+  }
+
+  public init?(bridged: BridgedSourceLoc) {
```

Comment:
this shouldn't be a public API

Reply:
In future, in other modules (i.e. `Parse`), we want to receive`Basic`'s `Bridged*` from C++
Like:

```swift
//-- Parse/Regex.swift
import Basic

// To be called from C++.
func parseRegexLiteral(_ bridgedLoc: BridgedSourceLoc, _ length: Int, ....) {
  let sourceLoc = SourceLoc(bridged: bridgedLoc)
  ...
}
```
So `init(bridged:)` needs to be `public`.